distribution/lib/Standard/Database/0.0.0-dev/src/Dialect.enso

```diff
@@ -269,6 +269,14 @@ type Dialect
         _ = [base_table, key_columns, resolved_aggregates, problem_builder]
         Unimplemented.throw "This is an interface only."
 
+    ## PRIVATE
+       Generates a column for the given expression for use in the SELECT clause. 
+       Used for databases where the expression sytanx is different in the SELECT clause 
+       to the synatx in the WHERE clause
```

Comment:
```suggestion
       Used for databases where the expression syntax is different in the SELECT clause 
       to the syntax in the WHERE clause
```

Reply:
What are the possible differences? 

What do you think about an alternative approach where we keep one `generate_expression` function and make it take an additional argument `purpose = Select | Where`?

Reply:
I think that could be a good approach. Any objections to doing it in the next MR?

Reply:
I feel like after all this is outdated overall. If we are on the same page after our discussions, I don't really think we need a separate expression generator for Select and Where contexts. 

Instead, we need to be careful in SQL Server to convert between 'booleans' and 'regular values' in the flow. So I think we need one expression generator, but we need a way to track the types of each operation's input and output and add conversions accordingly. 

Then we just need to keep in mind that the Select context takes a 'regular value' and Where takes a 'boolean expression' and tell the generator about these expectations to ensure that it does any conversions if necessary.